editor/import/resource_importer_wav.cpp

```diff
@@ -197,6 +197,8 @@ Error ResourceImporterWAV::import(const String &p_source_file, const String &p_s
 				break;
 			}
 
+			ERR_FAIL_COND_V_MSG(chunksize > INT_MAX, ERR_INVALID_DATA, "WAV file '" + p_source_file + "' is too large (> 2 GiB of data). Try using a compressed format such as Ogg Vorbis or MP3 instead.");
```

Comment:
Or try aplit the resource and shorten the audio length?

Reply:
We could truncate the stream, but I'm not sure this would benefit anyone. I can think of scenarios where such a huge file might end up in the project tree, but I can't think of any where you'd actually want Godot to import it. Doing a partial import of such a file might just produce gigabytes of unused data in the exported game.

Reply:
Oh sorry, I mean we can suggest the user to truncate the stream manually instead of import it as a whole.

Reply:
@bs-mwoerner: you got an "OK". I guess we can continue with getting this merged?

Maybe we can add something to the docs of `AudioStream` (or `AudioStreamWAV`), that audio file must be <= 4 GiB ?!

Reply:
@MJacred You're right, I missed the "thumbs up". I've updated the commit with the new wording.

The limit is actually 2 GiB of actual data, because the limiting factor here isn't the WAV format itself but the way Godot stores binary resources, so the stream after importing can't be longer than INT_MAX bytes. We can add that to the docs (I've never comitted to the documentation before but there's a first time for everything.) Would it make sense to include the doc update in this pull request?

Reply:
Godot can do more the int max

Reply:
It's been a while since I originally checked this, but I may have stopped looking prematurely when I saw that ` ResourceFormatSaverBinaryInstance::write_variant()` treated sizes > 2 GiB as negative lengths. At second glance, it seems to mostly call methods that reinterpret this as an unsigned int, so the overflows should cancel out. In that case, the problem may indeed just be the WAV stream itself.

Reply:
> Would it make sense to include the doc update in this pull request?

Yes :)